setup.py

```diff
@@ -123,6 +123,10 @@
     'AliPCS-Py>=0.8,<1',
 ]
 
+extra_deps['hf'] = [
+    'huggingface_hub>=0.23.4,<0.24',
```

Comment:
Can it be more loose? As it conflicts with our stack llmfoundry's [install requirements](https://github.com/mosaicml/llm-foundry/blob/main/setup.py) ('huggingface-hub>=0.19.0,<0.23',).

Reply:
I think it could be slightly looser, but not much.  Seems like it was [introduced in 0.21.0](https://github.com/huggingface/huggingface_hub/releases?page=2) and has had bug fixes since. [v0.23.0](https://github.com/huggingface/huggingface_hub/releases/tag/v0.23.0) also had a lot of changes to how downloads work including breaking API changes IIRC.  Probably why the conflict is there, because of all the changes.

Reply:
@XiaohanZhangCMU what are your thoughts on this?